### PR TITLE
IDEX IG velocity estimate limiter

### LINKED ISSUE #3462: BUG - IDEX L2A Ion Grid Velocity Estimate are unbounded
### Description of the issue

Wall impacts produce charge ratios outside our calibrated range and produce IG velocity estimates that are non-physical.

### Steps to reproduce the issue

_No response_

### Expected vs Actual behavior

_No response_

### Code Snippet (If applicable)

```python
Code
```

### Additional notes, affected areas, and suggested fixes

_No response_

### PR DESCRIPTION
**Pull request by aldo9253**
# Change Summary

  ## Overview

Closes #3462 

  Restrict IDEX Ion Grid velocity estimates to the inclusive range of 1–100 km/s. Estimates outside this range and their corresponding mass estimates are set to NaN. Fitted charges and waveform fits remain unchanged.

  ## File changes

  - idex_l2a.py: Add the velocity-range check before calculating Ion Grid mass.
  - test_idex_l2a.py: Add seven parameterized test cases covering excessive velocities with either target gain, inclusive boundaries, and values outside both limits.

  ## Testing

  - 110 IDEX tests passed.

### CODE REVIEW

File: imap_processing/idex/idex_l2a.py [hunk 1]
```diff
@@ -502,9 +504,13 @@ def calculate_ion_grid_velocity_and_mass(
         return np.nan, np.nan
 
     charge_ratio = ion_grid_charge / target_charge
+    if not 0.01 <= charge_ratio <= 1.0:
+        return np.nan, np.nan
```
[the remark this answers is not in the document]
Comment (lacoak21) [reply imap_processing/idex/idex_l2a.py:508]: Ah yes I was also wondering this. Maybe the comments are just off?

**Review by lacoak21 — COMMENTED** [review]
Looks good but there is still a pending comment. I will approve when its addressed.

File: imap_processing/idex/idex_l2a.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aldo9253) [reply imap_processing/idex/idex_l2a.py:508]: I had it wrong initially, the calibration fit was only run over a dataset for ratios from 1-100% not 1-100km/s.

**Review by lacoak21 — APPROVED** [review]
